evm/vm/message.py

```diff
@@ -142,3 +143,73 @@ def storage_address(self, value):
     @property
     def is_create(self):
         return self.to == CREATE_CONTRACT_ADDRESS
+
+
+class ShardingMessage(Message):
+
+    def __init__(self,
+                 gas,
+                 gas_price,
+                 to,
+                 sender,
+                 value,
+                 data,
+                 code,
+                 origin=None,
+                 access_list=None,
+                 depth=0,
+                 create_address=None,
+                 code_address=None,
+                 should_transfer_value=True,
+                 is_static=False):
+        validate_uint256(gas, title="Message.gas")
+        self.gas = gas
+
+        validate_uint256(gas_price, title="Message.gas_price")
+        self.gas_price = gas_price
+
+        validate_canonical_address(to, title="Message.to")
+        self.to = to
+
+        if sender != ENTRY_POINT:
+            validate_canonical_address(sender, title="Message.sender")
+        self.sender = sender
+
+        validate_uint256(value, title="Message.value")
```

Comment:
I think you can eliminate the majority of the body of this function by inserting a `super` call to the `Messag.__init__` method and then only dealing with what looks like the *only* difference which is `access_list` here in this method.

Reply:
ah yeah you are right, I thought they differ in how they handle `to` and `sender` but the effects are actually the same.